src/update-feeds/check-feeds.ts

```diff
@@ -0,0 +1,198 @@
+import { getState } from '../state';
+import type { ChainId, ProviderName } from '../types';
+import { multiplyBigNumber } from '../utils';
+
+import { getApi3ServerV1 } from './api3-server-v1';
+import type { ReadDapiWithIndexResponse } from './dapi-data-registry';
+import { decodeBeaconValue } from './update-feeds';
+import type { UpdatableDapi } from './update-transactions';
+
+export const shallowCheckFeeds = (
+  batch: ReadDapiWithIndexResponse[],
+  deviationThresholdCoefficient: DeviationThresholdCoefficient
+): UpdatableDapi[] =>
+  batch
+    .map((dapiInfo): UpdatableDapi | null => {
+      const beaconsSignedData = dapiInfo.decodedDataFeed.beacons.map((beacon) => getStoreDataPoint(beacon.beaconId));
+
+      // Only update data feed when we have signed data for all constituent beacons.
+      if (beaconsSignedData.some((signedData) => !signedData)) return null;
+      const beaconsDecodedValues = beaconsSignedData.map((signedData) => decodeBeaconValue(signedData!.encodedValue));
+      // Only update data feed when all beacon values are valid.
+      if (beaconsDecodedValues.includes(null)) return null;
+
+      // https://github.com/api3dao/airnode-protocol-v1/blob/fa95f043ce4b50e843e407b96f7ae3edcf899c32/contracts/api3-server-v1/DataFeedServer.sol#L163
+      const newBeaconSetValue = calculateMedian(beaconsDecodedValues.map((decodedValue) => decodedValue!));
+      const newBeaconSetTimestamp = calculateMedian(
+        beaconsSignedData.map((signedData) => ethers.BigNumber.from(signedData!.timestamp))
+      )!.toNumber();
+      const adjustedDeviationThresholdCoefficient = multiplyBigNumber(
+        dapiInfo.updateParameters.deviationThresholdInPercentage,
+        deviationThresholdCoefficient
+      );
+      if (
+        !checkUpdateConditions(
+          dapiInfo.dataFeedValue.value,
+          dapiInfo.dataFeedValue.timestamp,
+          newBeaconSetValue,
+          newBeaconSetTimestamp,
+          dapiInfo.updateParameters.heartbeatInterval,
+          adjustedDeviationThresholdCoefficient
+        )
+      ) {
+        return null;
+      }
+
+      return {
+        dapiInfo,
+        UpdatableBeacons: zip(dapiInfo.decodedDataFeed.beacons, beaconsSignedData).map(([beacon, signedData]) => ({
+          signedData: signedData!,
+          beaconId: beacon!.beaconId,
+        })),
+      };
+    })
+    .filter((UpdatableDapi): UpdatableDapi is UpdatableDapi => UpdatableDapi !== null);
+
+interface OnChainValue {
+  beaconId: string;
+  onChainValue: { timestamp: ethers.BigNumber; value: ethers.BigNumber };
+}
+
+export const callAndParseMulticall = async (
+  batch: string[],
+  providerName: ProviderName,
+  chainId: ChainId
+): Promise<OnChainValue[]> => {
+  const { config } = getState();
+  const chain = config.chains[chainId]!;
+  const { providers, contracts } = chain;
+
+  const provider = new ethers.providers.StaticJsonRpcProvider(providers[providerName]);
```

Comment:
We could probably pass the `provider` in as an input, but not a big deal either way 

I'd say don't do that here, I'll do it in my PR when I have to rebase on this PR since I have similar changes done there